core/src/main/java/hudson/search/UserSearchProperty.java

```diff
@@ -11,7 +11,9 @@
 import org.kohsuke.stapler.export.Exported;
 
 public class UserSearchProperty extends hudson.model.UserProperty {
-    
+
+    private static final boolean CASE_INSENSITIVE_SEARCH_MODE = true;
```

Comment:
Maybe `DEFAULT_SEARCH_ID_CASE_INSENSITIVE`? If you add such static field, maybe it is useful to initialize the default value using `SystemProperties.getBoolean()`